slp2mp4/slp-to-mp4.py

```diff
@@ -40,10 +41,91 @@
+
+
+combined_files = []
+
 
 def is_game_too_short(num_frames, remove_short):
     return num_frames < MIN_GAME_LENGTH and remove_short
 
+
+def record_file_slp(slp_file, outfile):
+    conf = Config()
+
+    # Parse file with py-slippi to determine number of frames
+    slippi_game = Game(slp_file)
+    num_frames = slippi_game.metadata.duration + DURATION_BUFFER
+
+    if is_game_too_short(num_frames, conf.remove_short):
+        print("Warning: Game is less than 30 seconds and won't be recorded. Override in config.")
+        return
+
+    # Dump frames
+    with DolphinRunner(conf, THIS_USER_DIR, SCRIPT_DIR, JOB_ID) as dolphin_runner:
+        video_file, audio_file = dolphin_runner.run(slp_file, num_frames)
+
+        # Encode
+        ffmpeg_runner = FfmpegRunner(conf.ffmpeg)
+        ffmpeg_runner.run(video_file, audio_file, outfile)
+
+        print('Created {}'.format(outfile))
+
+
+def combine(conf):
+    if not conf.combine:
+        return
+    for subdir, dirs, files in os.walk(OUT_DIR):
+        basedir = os.path.basename(subdir)
+        if os.path.exists(os.path.join(subdir, 'concat_file.txt')):
+            os.remove(os.path.join(subdir, 'concat_file.txt'))
+        file_count = 0
+        with open(os.path.join(subdir, 'concat_file.txt'), 'w+') as concat_file:
+            lines = []
+            for file in files:
+                try:
+                    if file.split('.')[-1] == 'mp4' and os.path.join(subdir, file) not in combined_files:
+                        file_count = file_count + 1
+                        lines.append("file \'" + os.path.join(subdir, file) + "\'" + "\n")
+                except Exception:
+                    pass
+            concat_file.writelines(lines)
+        if file_count > 0 and not os.path.exists(os.path.join(OUT_DIR, basedir) + '.mp4'):
+            ffmpeg_runner = FfmpegRunner(conf.ffmpeg)
+            ffmpeg_runner.combine(os.path.join(subdir, 'concat_file.txt'), os.path.join(OUT_DIR, basedir) + '.mp4')
+            combined_files.append(os.path.join(OUT_DIR, basedir) + '.mp4')
+            if os.path.exists(os.path.join(OUT_DIR, basedir)) and os.path.exists(os.path.join(OUT_DIR, basedir) + '.mp4'):
+                shutil.rmtree(os.path.join(OUT_DIR, basedir))
+        if os.path.exists(os.path.join(subdir, 'concat_file.txt')):
+            os.remove(os.path.join(subdir, 'concat_file.txt'))
+
+
+def record_folder_slp(slp_folder, conf):
+    in_files = []
+    out_files = []
+    for subdir, dirs, files in os.walk(slp_folder):
+        for file in files:
+            if file.split('.')[-1] == 'slp':
+                in_files.append([subdir, file])
+                out_files.append([os.path.basename(subdir), str(file.split('.')[:-1][0]) + '.mp4'])
+
+    if len(out_files) == 0:
+        RuntimeError("No slp files in folder!")
+    last_dir = out_files[0][0]
+    for index, in_file in enumerate(in_files):
+        if out_files[index][0] != last_dir:
+            combine(conf)
```

Comment:
I'm not sure why combine() is being called here - it traverses the entire out directory, so shouldn't it only be called at the end?

Reply:
I thought it would be better to combine as soon as a subdirectory was ready. Then, for example, you can start uploading to youtube part way through recording. Perhaps combine() should only traverse one specified subdirectory at a time given by record_folder_slp(). Do you think it is worth refactoring to do that?

Reply:
Oh I see, I didn't understand that intention.
To be honest for combine, I think you should limit the defined behavior to the most common/intended directory structure, and just describe that clearly with no guarantees on the behavior outside that case.
Also, in general for code niceness/modularity yes I think the combine() function should take a directory as an argument, but it might not be necessary for this PR.